generic/servicelog.py

```diff
@@ -0,0 +1,269 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+# Copyright: 2016 IBM
+# Authors: Rafael Camarda Silva Folco (rfolco@br.ibm.com) (Original author)
+#          Lucas Meneghel Rodrigues (lucasmr@br.ibm.com)
+#          Hariharan T.S. <hari@linux.vnet.ibm.com> (ported to avocado)
+#          Pavithra <pavrampu@linux.vnet.ibm.com> (Added test_servicelog)
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import genio
+from avocado.utils.service import ServiceManager
+from avocado.utils.software_manager import SoftwareManager
+
+
+class servicelog(Test):
+
+    is_fail = 0
+    EVENTS_PATH = os.path.join(os.path.dirname(__file__), "servicelog.py.data", "rtas_events")
+
+    def run_cmd(self, cmd):
+        if process.system(cmd, ignore_status=True, sudo=True, shell=True):
+            self.is_fail += 1
+        return
+
+    def setUp(self):
+        if "ppc" not in os.uname()[4]:
+            self.skip("supported only on Power platform")
+        sm = SoftwareManager()
+        if 'PowerNV' in open('/proc/cpuinfo', 'r').read():
+            self.skip("servicelog: is not supported on the PowerNV platform")
+        for package in ("servicelog", "ppc64-diag"):
+            if not sm.check_installed(package) and not sm.install(package):
+                self.skip("Fail to install %s required for this"
+                          " test." % package)
+
+    def test_servicelog(self):
+        """
+        The test checks servicelog commands.
+        """
+        rtas_events = ["v6_fru_replacement", "v6_memory_info", "v6_power_error",
+                       "v6_power_error2", "v3_predictive_cpu_failure",
+                       "v6_predictive_cpu_failure", "v6_fw_predictive_error",
+                       "v4_io_bus_failure", "v6_io_sub_error"]
+        for event in rtas_events:
+            self.log.info("Starting test scenario for %s" % event)
+            self.log.info("1 - Cleaning servicelog...")
+            self.run_cmd("servicelog_manage --truncate notify --force")
+            self.log.info("2 - Injecting event %s" % event)
+            self.run_cmd("rtas_errd -d -f %s/%s" % (self.EVENTS_PATH, event))
+            self.log.info("3 - Checking if service log does return ppc64_rtas events")
+            self.run_cmd("servicelog --type=ppc64_rtas")
+            inject = process.system_output("servicelog --type=all | "
+                                           "grep 'PPC64 Platform Event' | cut -d':' -f1", shell=True)
+            if "PPC64 Platform Event" == inject:
+                self.is_fail += 1
+                self.log.info("Event %s error: Event generated is not a "
+                              "'PPC64 Platform Event'" % event)
```

Comment:
Since we are only including the list of events, can't we just have only "PPC64 Platform Event"s in the list?